CMakeLists.txt

```diff
@@ -38,7 +48,117 @@ mxnet_option(BUILD_CPP_EXAMPLES   "Build cpp examples" ON)
+#   community support licensing (https://software.intel.com/en-us/articles/free-mkl),
+#   but needs to be downloaded and installed manually.
+#
+# * MKLML is a subset of MKL. It contains smaller number of functions to reduce the
+#   size of the download and reduce the number of dynamic libraries user needs. This
+#   is the most effective option since it can be downloaded and installed automatically
+#   by the cmake script (see cmake/DownloadMKLML.cmake).
+#
+# * MKLDNN is a separate open-source library, but it requires MKL or MKLML. It is
+#   shipped as a subrepo with MXNet source code (see 3rdparty/mkldnn).
+#
+# Since the full MKL library is almost always faster than any other it's turned on by default,
+# however it needs to be downloaded and installed manually before doing cmake configuration.
+# Register and download here https://software.seek.intel.com/performance-libraries
+#
+# Note: MKL is supported only for desktop builds and the framework itself supports the following
+# hardware:
+#
+# * Intel® Xeon Phi™ processor
+# * Intel® Xeon® processor
+# * Intel® Core™ processor family
+# * Intel Atom® processor
+#
+# If you have a different processor you can still try to use MKL, but performance results are
+# unpredictable.
+mxnet_option(USE_MKL_IF_AVAILABLE "Use MKL if found" ON)
+
+# If the full MKL library could not be found the thinner subset MKLML will be downloaded
+# unless switched off explicitly.
+# Note: The same limitation on hardware as for MKL applies for MKLML as well.
+mxnet_option(USE_MKLML "Use MKLML subset of MKL instead of mkl, will be downloaded" ON)
+
+# If either MKL of MKLML is present MKLDNN can be utilised from the 3rdparty/mkldnn subrepo.
+# See more information here: https://github.com/intel/mkl-dnn
+# Note: The same limitation on hardware as for MKL and MKLDNN applies for MKLDNN as well.
+mxnet_option(USE_MKLDNN "Use MKLDNN (requires MKLML or MKL)" ON)
+
+# Apple's mathematical framework, probably the best choice on a Mac if MKL/MKLML/MKLDNN
+# are not available.
+# https://developer.apple.com/documentation/accelerate
+mxnet_option(USE_APPLE_ACCELERATE_IF_AVAILABLE "Use Apple Accelerate framework if found, \
+                                                works if MKL not found or disabled" ON IF ${APPLE})
+
+# Another important option of the math libraries is presence of additional set of
+# mathematical functions gathered and named as the LAPACK (Linear Algebra Package). Some
+# libraries don't include it, thus the cmake script will check the presence of an
+# indicating function "cheev_" within the available choosen libraries and switch the
+# functionality off if not found.
+mxnet_option(USE_LAPACK "Build with lapack support" ON)
+
+if(NOT mxnet_LINKER_LIBS)
+  set(mxnet_LINKER_LIBS "")
+endif(NOT mxnet_LINKER_LIBS)
+
+if(MSVC)
+  set(SYSTEM_ARCHITECTURE x86_64)
+else()
+  execute_process(COMMAND uname -m COMMAND tr -d '\n' OUTPUT_VARIABLE SYSTEM_ARCHITECTURE)
```

Comment:
What if we are cross compiling?

Reply:
That supposed to be the host architecture. It's used to not try to download MKLML in case we are on an arm platform.

Reply:
But we cross compile for ARM in the host.

Reply:
For cross compilation we use the CMAKE_CROSSCOMPILING flag to prevent a download